### ISSUE TITLE
Wrong HTTP status code on error

### ISSUE THREAD

**Issue by AirPett** [microsoftgraph/microsoft-graph-docs#7128]
If I try to delete a non-existing device, the API returns `400 Bad Request` with a message `Cannot find device with ztd Id XXXX and accountId YYYY to delete`. It would be more consistent and intuitive to return a `404 Not Found` as the status code and then the message with it.

E.g. `GET /importedWindowsAutopilotDeviceIdentities/AAAA` and `GET /windowsAutopilotDeviceIdentities/BBBB` properly return `404 Not Found` when a non-existent device is queried.

From the current message where the only place it says "Cannot find" is in the Message, it's hard to parse the reason for the `400 Bad Request`. The proper error code would make it much easier to detect if a device was not found.

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: c893c037-62dd-3712-8441-2d74aab9fbcb
* Version Independent ID: db3f1e7a-26c3-a785-df23-35164fa54889
* Content: [Delete windowsAutopilotDeviceIdentity - Microsoft Graph beta](https://docs.microsoft.com/en-us/graph/api/intune-enrollment-windowsautopilotdeviceidentity-delete?view=graph-rest-beta)
* Content Source: [api-reference/beta/api/intune-enrollment-windowsautopilotdeviceidentity-delete.md](https://github.com/microsoftgraph/microsoft-graph-docs/blob/master/api-reference/beta/api/intune-enrollment-windowsautopilotdeviceidentity-delete.md)
* Product: **intune**
* Technology: **microsoft-graph**
* GitHub Login: @rolyon
* Microsoft Alias: **MSGraphDocsVteam**

**Comment by triplanedave** [comment]
Thanks, I've logged a work item with our team to update the error information.
